base/math.jl

```diff
@@ -537,38 +537,61 @@ Stacktrace:
 [...]
 ```
 """
-hypot(x::Number, y::Number) = hypot(promote(x, y)...)
-function hypot(x::T, y::T) where T<:Number
-    ax = abs(x)
-    ay = abs(y)
-    if ax < ay
-        ax, ay = ay, ax
-    end
+function hypot(x::Number, y::Number)
+    # Specialized two argument version is faster than generic varargs version
+    ay, ax = minmax(hypot(x), hypot(y))
     if iszero(ax)
         r = ay / oneunit(ax)
     else
         r = ay / ax
     end
-
-    rr = ax * sqrt(1 + r * r)
-
-    # Use type of rr to make sure that return type is the same for
-    # all branches
+    r2 = r^2
+    rr = ax * sqrt(oneunit(r2) + r2)
     if isnan(r)
-        isinf(ax) && return oftype(rr, Inf)
-        isinf(ay) && return oftype(rr, Inf)
-        return oftype(rr, r)
+        if isinf(x) || isinf(y)
+            return oftype(rr, Inf)
+        else
+            return oftype(rr, r)
+        end
     else
         return rr
-    end
+   	end
 end
 
 """
     hypot(x...)
 
 Compute the hypotenuse ``\\sqrt{\\sum x_i^2}`` avoiding overflow and underflow.
 """
-hypot(x::Number...) = sqrt(sum(abs2(y) for y in x))
+function hypot(x::Number...)
```

Comment:
It may be possible to avoid the need for `hypot()` by defining this as `hypot(x::Number, xs::Number...)`. 

Also, `x` isn't a good variable name for a tuple (since it is used above as the variable name of a scalar.